airbyte-integrations/connectors/source-pinterest/source_pinterest/components/record_filter.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Optional
+
+import pendulum
+from airbyte_cdk.sources.declarative.extractors.record_filter import RecordFilter
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+
+
+class PinterestSemiIncrementalRecordFilter(RecordFilter):
```

Comment:
similar to above, a comment to explain the need for a custom component

Reply:
Added